vendor/bat-native-ledger/src/bat/ledger/internal/media/github.cc

```diff
@@ -0,0 +1,339 @@
+              visit_data,
+              publisher_key,
+              _1,
+              _2));
+}
+
+
+
+void GitHub::OnPublisherPanelInfo(
+    uint64_t window_id,
+    const ledger::VisitData& visit_data,
+    const std::string& publisher_key,
+    ledger::Result result,
+    ledger::PublisherInfoPtr info) {
+  if (!info || result == ledger::Result::NOT_FOUND) {
+    FetchDataFromUrl(visit_data.url,
+                     std::bind(&GitHub::OnUserPage,
+                               this,
+                               window_id,
+                               visit_data,
+                               _1,
+                               _2,
+                               _3));
+  } else {
+    ledger_->OnPanelPublisherInfo(result, std::move(info), window_id);
+  }
+}
+
+void GitHub::FetchDataFromUrl(
+    const std::string& url,
+    braveledger_media::FetchDataFromUrlCallback callback) {
+  ledger_->LoadURL(url,
+                   std::vector<std::string>(),
+                   std::string(),
+                   std::string(),
+                   ledger::URL_METHOD::GET,
+                   callback);
+}
+void GitHub::OnUserPage(
+    uint64_t window_id,
+    const ledger::VisitData& visit_data,
+    int response_status_code,
+    const std::string& response,
+    const std::map<std::string, std::string>& headers) {
+
+  if (response_status_code != net::HTTP_OK) {
+    OnMediaActivityError(visit_data, window_id);
+    return;
+  }
+
+  const std::string user_id = GetUserId(response);
+  const std::string user_name = GetUserNameFromUrl(visit_data.path);
+  const std::string publisher_name = GetPublisherName(response);
+  const std::string profile_picture = GetProfileImageURL(response);
+
+  auto callback = std::bind(&GitHub::OnSaveMediaVisit,
+                            this,
+                            _1,
+                            _2);
+
+  SavePublisherInfo(0,
+                    user_id,
+                    user_name,
+                    publisher_name,
+                    profile_picture,
+                    window_id,
+                    callback);
+}
+
+void GitHub::OnSaveMediaVisit(
+    ledger::Result result,
+    ledger::PublisherInfoPtr info) {
+    // jkuki TODO handle if needed
+}
+
+void GitHub::SavePublisherInfo(
+    const uint64_t duration,
+    const std::string& user_id,
+    const std::string& screen_name,
+    const std::string& publisher_name,
+    const std::string& profile_picture,
+    const uint64_t window_id,
+    ledger::PublisherInfoCallback callback) {
+  const std::string publisher_key = GetPublisherKey(user_id);
+  const std::string url = GetProfileURL(screen_name);
+  const std::string media_key = GetMediaKey(screen_name);
+
+  if (publisher_key.empty()) {
+    callback(ledger::Result::LEDGER_ERROR, nullptr);
+    BLOG(ledger_, ledger::LogLevel::LOG_ERROR) <<
+      "Publisher key is missing for: " << media_key;
+    return;
+  }
+
+  ledger::VisitDataPtr visit_data = ledger::VisitData::New();
```

Comment:
you can just create regular object here, so that you don't need to convert it on line 326